### PR TITLE
implement wasm test case framework

### PR DESCRIPTION
**Pull request by laizy**

### CODE REVIEW

File: core/store/ledgerstore/ledger_store.go [hunk 1]
```diff
@@ -1028,10 +1028,15 @@ func (this *LedgerStoreImp) GetEventNotifyByBlock(height uint32) ([]*event.Execu
 //PreExecuteContract return the result of smart contract execution without commit to store
 func (this *LedgerStoreImp) PreExecuteContract(tx *types.Transaction) (*sstate.PreExecResult, error) {
 	height := this.GetCurrentBlockHeight()
+	// use previous block time to make it predictable for easy test
+	blockTime := uint32(time.Now().Unix())
+	if header, err := this.GetHeaderByHeight(height); err == nil {
+		blockTime = header.Timestamp + 1
+	}
```
Comment (carltraveler) [inline core/store/ledgerstore/ledger_store.go:1035]: add err assert here?

File: wasmtest/run-wasm-tests.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+set -e
+set -x
+
+# install build tools
+if ! which rustup ; then
+	curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain nightly 
+	source $HOME/.cargo/env
+fi
+rustup target add wasm32-unknown-unknown
+which ontio-wasm-build || cargo install --git=https://github.com/ontio/ontio-wasm-build
```
Comment (carltraveler) [inline wasmtest/run-wasm-tests.sh:11]: this code setup move it to contracts-rust better?

Comment (laizy) [reply wasmtest/run-wasm-tests.sh:11]: `ontio-wasm-build` is used below, I prefer group the setup and usage   together.

File: core/store/ledgerstore/ledger_store.go (hunk 1 above)
Comment (laizy) [reply core/store/ledgerstore/ledger_store.go:1035]: why need assert here? `err` could be not `nil`, in this case we fall back to use `time.Now`.

**Review by carltraveler — APPROVED** [review]

**Pull request merged**